beater/beater.go

```diff
@@ -569,8 +569,8 @@ func (s *serverRunner) waitReady(ctx context.Context, kibanaClient kibana.Client
 }
 
 // newFinalBatchProcessor returns the final model.BatchProcessor that publishes events.
-func (s *serverRunner) newFinalBatchProcessor(libbeatReporter publish.Reporter) model.BatchProcessor {
-	return &reporterBatchProcessor{libbeatReporter}
+func (s *serverRunner) newFinalBatchProcessor(p *publish.Publisher) model.BatchProcessor {
+	return p
 }
```

Comment:
```suggestion
```
Do we need to keep this function? We can just remove it since it doesn't do anything anymore, right? it's just returning the same argument that's passed back to the caller.

Reply:
I intend to rebase https://github.com/elastic/apm-server/pull/5970 after this merges, at which point it'll be non-trivial. Depending on the configuration, `newFinalBatchProcessor` will either return a libbeat or go-elasticsearch processor.